src/VisualStudio/Core/Def/StackTraceExplorer/StackTraceExplorerCommandHandler.cs

```diff
@@ -107,16 +108,18 @@ private void UnadviseBroadcastMessages()
 
     private void GlobalOptionChanged(object sender, OptionChangedEventArgs e)
     {
-        if (e.Option == StackTraceExplorerOptionsStorage.OpenOnFocus && e.Value is not null)
+        foreach (var (key, newValue) in e.ChangedOptions)
         {
-            var enabled = (bool)e.Value;
-            if (enabled)
+            if (key.Option.Equals(StackTraceExplorerOptionsStorage.OpenOnFocus) && newValue is bool enabled)
             {
-                AdviseBroadcastMessages();
-            }
-            else
-            {
-                UnadviseBroadcastMessages();
+                if (enabled)
+                {
+                    AdviseBroadcastMessages();
+                }
+                else
+                {
+                    UnadviseBroadcastMessages();
+                }
```

Comment:
i'm wary of this loop.  shoudl we compute the value once in the loop, and then advise/unadvise once after instead?

Reply:
In practice the options are unique, but I guess that's not guaranteed anywhere and they might repeat.

Reply:
ok.  as long as you're ok with this going on/off multiple times, that's ok with me.  i genuinely dont' know what the semantics are ere.